rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_participant_info.hpp

```diff
@@ -73,22 +73,18 @@ class ParticipantListener : public eprosima::fastrtps::ParticipantListener
       case eprosima::fastrtps::rtps::ParticipantDiscoveryInfo::DISCOVERED_PARTICIPANT:
         {
           auto map = rmw::impl::cpp::parse_key_value(info.info.m_userData);
-          auto name_found = map.find("name");
-          auto ns_found = map.find("namespace");
+          auto name_found = map.find("contextname");
```

Comment:
@ivanpauno nit: 
```suggestion
          auto name_found = map.find("context_name");
```

Reply:
I started using that. The function parsing the key doesn't accept an underscore in it :rofl::
https://github.com/ros2/rmw/blob/f76511333709a6f5d1e4543129ec73f2cd2b4f06/rmw/include/rmw/impl/cpp/key_value.hpp#L70

After I realized the root cause of the bug I was experimenting, I just dropped the underscore from the key name.

If you want, I can fix `parse_key`.

Reply:
I realized this wasn't  in sync with https://github.com/ros2/rmw_fastrtps/blob/f10cc19644cb1763d7b9db2d8cecc42e6f3a4051/rmw_fastrtps_shared_cpp/src/participant.cpp#L174

Fixed in https://github.com/ros2/rmw_fastrtps/pull/312/commits/f10cc19644cb1763d7b9db2d8cecc42e6f3a4051.